Orange/widgets/visualize/owfreeviz.py

```diff
@@ -370,9 +370,10 @@ def init_embedding_coords(self):
     def get_embedding(self):
         if self.data is None:
             return None
-        embedding = np.dot(self._X, self.projection)
-        embedding /= \
-            np.max(np.linalg.norm(embedding[self.valid_data], axis=1)) or 1
+        EX = np.dot(self._X, self.projection)
+        EX /= np.max(np.linalg.norm(EX, axis=1)) or 1
+        embedding = np.zeros((len(self.data), 2), dtype=np.float)
```

Comment:
If the original data had missing values, and the embedding is not available for all points, I think it is better to assign missing values to those points as the embedding too (instead of zeros).

Reply:
Also, while I doubt these projections will ever be generalized to more than 2D, I would still prefer to not make implicit assumptions like that in the code and use `EX.shape[1]` instead of `2`